File: frontend/src/utils/sentiment.js
export const sentimentEmojis = {
  positive: "😊",
  neutral: "😐", 
  negative: "😞"
};

export const getSentimentEmoji = (sentiment) => {
  if (!sentiment || sentiment === 'neutral') return sentimentEmojis.neutral;
  if (sentiment === 'positive') return sentimentEmojis.positive;
  if (sentiment === 'negative') return sentimentEmojis.negative;
  return sentimentEmojis.neutral;
};

File: backend/users/sentiment.py
"""
Sentiment analysis utility using TextBlob.
Classifies feedback comments as positive, neutral, or negative.
"""
from textblob import TextBlob


def analyze_sentiment(text):
    """
    Analyze the sentiment of text using TextBlob.
    
    Returns:
        tuple: (polarity, label) where label is 'positive', 'neutral', or 'negative'
    """
    if not text or not text.strip():
        return 0.0, 'neutral'

    blob = TextBlob(text)
    polarity = blob.sentiment[0]

    if polarity > 0.1:
        label = 'positive'
    elif polarity < -0.1:
        label = 'negative'
    else:
        label = 'neutral'
        
    return round(float(polarity), 3), label


def get_sentiment_emoji(sentiment):
    """Return emoji for a given sentiment."""
    emojis = {
        'positive': '😊',
        'neutral': '😐',
        'negative': '😞',
    }
    return emojis.get(sentiment, '😐')
